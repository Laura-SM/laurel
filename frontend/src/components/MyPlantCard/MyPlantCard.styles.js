import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 175,
    height: 175,
    borderWidth: 2,
    borderColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
  },
  image: {
    width: 100,
    height: 100,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
});

export default styles;
