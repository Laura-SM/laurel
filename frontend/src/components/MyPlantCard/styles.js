import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 165,
    height: 230,
    backgroundColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginVertical: 10,
    marginLeft: 20,
  },
  image: {
    position: 'absolute',
    top: 0,
    width: 165,
    height: 165,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
  text: {
    fontSize: 18,
    marginHorizontal: 20,
    marginBottom: 10,
  },
});

export default styles;
