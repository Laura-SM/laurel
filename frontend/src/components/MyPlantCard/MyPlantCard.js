import React from 'react';
import {Image, Pressable, Text} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import styles from './MyPlantCard.styles';
import globalStyles from '../../styles/global.styles';

const MyPlantCard = ({plant}) => {
  const navigation = useNavigation();
  return (
    <Pressable
      style={styles.cardContainer}
      onPress={() =>
        navigation.navigate('MyPlantDetails', {plantId: plant._id})
      }>
      <Image style={styles.image} source={{uri: plant.image}} />
      <Text style={globalStyles.boldText}>{plant.name}</Text>
      <Text style={globalStyles.text}>{plant.room}</Text>
    </Pressable>
  );
};

export default MyPlantCard;
