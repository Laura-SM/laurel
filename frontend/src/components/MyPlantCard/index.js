import React from 'react';
import {Image, Pressable, Text} from 'react-native';
import styles from './styles';
import globalStyles from '../../styles/globalStyles';

const MyPlantCard = ({plant, navigation}) => {
  return (
    <Pressable
      style={styles.cardContainer}
      onPress={() =>
        navigation.navigate('MyPlantDetails', {plantId: plant._id})
      }>
      <Image style={styles.image} source={{uri: plant.image}} />
      <Text style={globalStyles.boldText}>{plant.name}</Text>
      <Text style={globalStyles.text}>{plant.room}</Text>
    </Pressable>
  );
};

export default MyPlantCard;
