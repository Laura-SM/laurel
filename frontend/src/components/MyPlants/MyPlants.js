/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect} from 'react';
import {Text, View, FlatList, TouchableOpacity, Image} from 'react-native';
import {connect} from 'react-redux';
import {useNavigation} from '@react-navigation/native';
import {loadPlants} from '../../redux/actions/plantsActionCreators';
import MyPlantCard from '../MyPlantCard/MyPlantCard';
import globalStyles from '../../styles/global.styles';

const MyPlants = ({plants, userAccess, dispatch}) => {
  const navigation = useNavigation();
  const myPlantsIds = [...userAccess.user.plants];
  const myPlantsList = [];
  if (myPlantsIds.length) {
    myPlantsIds.forEach(id => {
      let myPlant = plants.find(plant => plant._id === id);
      myPlantsList.push(myPlant);
    });
  }
  const renderMyPlantCard = ({item}) => <MyPlantCard plant={item} />;

  useEffect(() => {
    if (userAccess.user.plants.length) {
      dispatch(loadPlants(userAccess.user.plants));
    }
  }, [userAccess.user.plants]);

  return (
    <FlatList
      style={globalStyles.mainContainer}
      ListHeaderComponent={
        <View style={globalStyles.headerContainer}>
          <Text style={globalStyles.titleText}>My plants</Text>
          <TouchableOpacity
            style={globalStyles.roundButton}
            onPress={() => navigation.navigate('AddPlant')}>
            <Image source={require('../../icons/add24.png')} />
          </TouchableOpacity>
        </View>
      }
      data={myPlantsList}
      renderItem={renderMyPlantCard}
      keyExtractor={plant => plant._id}
      horizontal={false}
      numColumns={2}
    />
  );
};

function mapStateToProps({userAccess, plants}) {
  return {
    userAccess,
    plants,
  };
}

export default connect(mapStateToProps)(MyPlants);
