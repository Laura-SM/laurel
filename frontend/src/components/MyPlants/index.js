import React from 'react';
import {Text, View, FlatList, TouchableOpacity, Image} from 'react-native';
import {connect} from 'react-redux';
import MyPlantCard from '../MyPlantCard';
import globalStyles from '../../styles/globalStyles';

const MyPlants = ({plants, userAccess, navigation}) => {
  const myPlantsIds = [...userAccess.user.plants];
  const myPlantsList = [];
  if (myPlantsIds.length) {
    myPlantsIds.forEach(id => {
      let myPlant = plants.find(plant => plant._id === id);
      myPlantsList.push(myPlant);
    });
  }
  const renderMyPlantCard = ({item}) => <MyPlantCard plant={item} />;

  return (
    <FlatList
      style={globalStyles.mainContainer}
      ListEmptyComponent={
        <Text style={globalStyles.text}>Add your plants!</Text>
      }
      ListHeaderComponent={
        <View style={globalStyles.headerContainer}>
          <Text style={globalStyles.titleText}>My plants</Text>
          <TouchableOpacity
            style={globalStyles.roundButton}
            onPress={() => navigation.navigate('AddPlant')}>
            <Image source={require('../../icons/add24.png')} />
          </TouchableOpacity>
        </View>
      }
      data={myPlantsList}
      renderItem={renderMyPlantCard}
      keyExtractor={plant => plant._id}
      horizontal={false}
      numColumns={2}
    />
  );
};

function mapStateToProps({userAccess, plants}) {
  return {
    userAccess,
    plants,
  };
}

export default connect(mapStateToProps)(MyPlants);
