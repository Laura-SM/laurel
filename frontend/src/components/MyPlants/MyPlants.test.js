import React from 'react';
import MyPlants from './index';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import MyPlantCard from '../MyPlantCard';

describe('Given a MyPlants component', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    plants: [
      {name: 'kentia', _id: '1a'},
      {name: 'potus', _id: '2a'},
    ],
    userAccess: {
      user: {
        plants: ['1a'],
      },
    },
  });
  const navigation = {
    navigate: jest.fn(),
  };

  describe('When is invoked', () => {
    test('Then it is renders correctly', () => {
      const wrapper = render(
        <Provider store={store}>
          <MyPlants>
            <MyPlantCard />
          </MyPlants>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
