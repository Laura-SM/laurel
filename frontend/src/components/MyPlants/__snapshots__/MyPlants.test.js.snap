// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MyPlants component When is invoked Then it is renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginHorizontal": 20,
        }
      }
    >
      Add your plants!
    </Text>
  }
  ListHeaderComponent={
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 20,
          }
        }
      >
        My plants
      </Text>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#B8E0D2",
            "borderRadius": 50,
            "height": 60,
            "justifyContent": "center",
            "margin": 20,
            "width": 60,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/icons/add24.png",
            }
          }
        />
      </ForwardRef>
    </View>
  }
  data={
    Array [
      Object {
        "_id": "1a",
        "name": "kentia",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "margin": 20,
            }
          }
        >
          My plants
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B8E0D2",
              "borderRadius": 50,
              "height": 60,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/icons/add24.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#93B1A7",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 165,
              "justifyContent": "center",
              "marginLeft": 20,
              "marginVertical": 10,
              "width": 165,
            }
          }
          testID="goToDetails"
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "backgroundColor": "#E5E5E5",
                "borderRadius": 10,
                "height": 100,
                "width": 100,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginHorizontal": 20,
              }
            }
          >
            kentia
          </Text>
          <Text
            style={
              Object {
                "fontSize": 18,
                "marginHorizontal": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
